python/graphstorm/model/utils.py

```diff
@@ -1359,29 +1394,118 @@ def load_sparse_emb(target_sparse_emb, ntype_emb_path):
     rank = get_rank()
     world_size = get_world_size()
     num_files = len(os.listdir(ntype_emb_path))
-    num_embs = target_sparse_emb.num_embeddings
+    if is_wholegraph_sparse_emb() and is_wholegraph_embedding_module(target_sparse_emb):
+        num_embs = target_sparse_emb.wm_embedding.shape[0]
+    else:
+        num_embs = target_sparse_emb.num_embeddings
+
+    def _wholegraph_load_scatter(num_embs, file_id, num_files, file_path):
+        def _get_sparse_emb_lowerbound_range(file_size, rank, world_size):
```

Comment:
I think tools/convert_feat_to_wholegraph.py uses a similar logic. Shall we unify the implementation and move this _get_sparse_emb_lowerbound_range to wholegraph.py as a helper function?

Reply:
Agree.

Reply:
I simplified logic to consistent with tools/convert_feat_to_wholegraph.py.
The downside is: the saved sparse embs are in binary format, i.e., meta info, like shapes, dtype etc.  will not be available. It might not be an issue, as when we load it via `load_sparse_emb()`, the meta info should be available in `target_sparse_emb`, right?